nostr/subscription.py

```diff
@@ -1,8 +1,8 @@
 from .filter import Filters
 
 class Subscription:
-    def __init__(self, id: str, filters: Filters=None) -> None:
-        self.id = id
+    def __init__(self, id: str, filters: Filters) -> None:
+        self.id = str(id)
```

Comment:
I think it could be confusing to someone if we just coerce the id to a string without their knowledge. Instead, I suggest we raise an exception similar to [Event](https://github.com/jeffthibault/python-nostr/blob/main/nostr/event.py#L29-L30). This way, if someone tries to make an id that isn't a string, they get an explanation.